crates/rooch-framework/sources/nft/collection.move

```diff
@@ -0,0 +1,291 @@
+        let collection_object_mut_ref = context::borrow_object_mut<Collection>(ctx, mutator_object_ref.collection);
+        let collection_mut_ref = object::borrow_mut(collection_object_mut_ref);
+        collection_mut_ref.supply.current = collection_mut_ref.supply.current + 1;
+        if(option::is_some(&collection_mut_ref.supply.maximum)){
+            assert!(collection_mut_ref.supply.current <= *option::borrow(&collection_mut_ref.supply.maximum), ErrorCollectionMaximumSupply);
+            option::some(collection_mut_ref.supply.current)
+        }else{
+            option::none<u64>()
+        }
+    }
+
+    public (friend) fun decrement_supply(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context): Option<u64>{
+        assert_mutator_exist_of_ref(mutator);
+        let mutator_object_ref = object_ref::borrow(mutator);
+        assert_collection_exist_of_id(mutator_object_ref.collection, ctx);
+        let collection_object_mut_ref = context::borrow_object_mut<Collection>(ctx, mutator_object_ref.collection);
+        let collection_mut_ref = object::borrow_mut(collection_object_mut_ref);
+        collection_mut_ref.supply.current = collection_mut_ref.supply.current - 1;
+        if(option::is_some(&collection_mut_ref.supply.maximum)){
+            option::some(collection_mut_ref.supply.current)
+        }else{
+            option::none<u64>()
+        }
+    }
+
+    // assert
+    public fun assert_collection_exist_of_ref(collectionRef: &ObjectRef<Collection>){
+        assert!( object_ref::exist_object(collectionRef), ErrorCollectionNotExist);
+    }
+
+    public fun assert_collection_exist_of_id(collectionID: ObjectID, ctx: & Context){
+        assert!( context::exist_object(ctx, collectionID), ErrorCollectionNotExist);
+        context::borrow_object<Collection>(ctx,collectionID);
+    }
+
+    public fun assert_mutator_exist_of_ref(mutatorRef: &ObjectRef<MutatorRef>){
+        assert!( object_ref::exist_object(mutatorRef), ErrorMutatorNotExist);
+    }
+
+    public fun assert_mutator_exist_of_id(mutatorID: ObjectID, ctx: & Context){
+        assert!( context::exist_object(ctx, mutatorID), ErrorMutatorNotExist);
+        context::borrow_object<MutatorRef>(ctx, mutatorID);
+    }
+
+    public fun add_display(mutator: &ObjectRef<MutatorRef>, display: Display, ctx: &mut Context){
+        add_extend_internal(mutator, display, ctx);
+    }
+
+    public fun borrow_display(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context):&Display{
+        borrow_extend_internal(mutator, ctx)
+    }
+
+    public fun borrow_mut_display(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context):&mut Display{
+        borrow_mut_extend_internal(mutator, ctx)
+    }
+
+    public fun remove_display(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context):Display{
+        remove_extend_internal(mutator, ctx)
+    }
+
+    public fun contains_display(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context): bool{
```

Comment:
Not all functions need to pass in ObjectRef<MutatorRef>. For example, the `contains` function can pass in ObjectRef<Collection>?

Reply:
> Not all functions need to pass in ObjectRef. For example, the `contains` function can pass in ObjectRef?

This has been changed to struct because there are still some problems with the current use of Object